frontend/packages/operator-lifecycle-manager/src/components/descriptors/utils.tsx

```diff
@@ -11,6 +11,23 @@ import {
   REGEXP_NESTED_ARRAY_PATH,
 } from './const';
 import { Descriptor, SpecCapability, StatusCapability } from './types';
+import { getSchemaAtPath } from '@console/shared';
+
+export const withCalculatedDescriptorProperties = (descriptorType, descriptor, schema, obj) => {
```

Comment:
The `with` prefix here makes this sound like it's a higher order component like `withFallback` or `withTranslation`, but it's not. I might just call this `getDescriptorProperties`